### PR TITLE
Hide much infrastructure in data

### PR DESCRIPTION
**Pull request by TomFinley**
Another of many steps towards #1602. Commits logically structured. No overall theme, just lots of hiding of individual components, most notably command line parsing, entry-point declarations, and other such things.

### CODE REVIEW

**Comment by sfilipi** [comment]
>         public const string TakeFilterShortName = "Take";

hurts my eyes :)

---
Refers to: src/Microsoft.ML.Data/Transforms/SkipTakeFilter.cs:48 in 3ac69ce. [](commit_id = 3ac69ce8539d613404094fa43984f7f31cb4ac47, deletion_comment = False)

**Comment by codemzs** [comment]
Can you please sync to the latest in master because I’m really curious what the new code coverage is?

**Comment by sfilipi** [comment]
I synched at 6:30, at the latest..

Trying to figure out why a test is failing atm, before going through the rebase exercise again. Keep you posted; I want to get to it preferably tonigh.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

________________________________

**Comment by codemzs** [comment]
@sfilipi Isn’t this @TomFinley ‘s PR? :)

File: src/Microsoft.ML.Data/Transforms/GenerateNumberTransform.cs [hunk 1]
```diff
@@ -31,7 +31,8 @@ namespace Microsoft.ML.Transforms
     /// It will be used in conjunction with a filter transform to create random
     /// partitions of the data, used in cross validation.
     /// </summary>
-    public sealed class GenerateNumberTransform : RowToRowTransformBase
+    [BestFriend]
+    internal sealed class GenerateNumberTransform : RowToRowTransformBase
```
[the remark this answers is not in the document]
Comment (TomFinley) [reply src/Microsoft.ML.Data/Transforms/GenerateNumberTransform.cs]: Pursuant to #1995, all `IDataTransform`s (as opposed to `IEstimator`/`ITransformer`s) are to be hidden. So when hiding the entry-point exposing this class, I happened to opportunistically hide this as well, while I was at it.

**Comment by TomFinley** [comment]
> Can you please sync to the latest in master because I’m really curious what the new code coverage is?

If it naturally comes up due to an actual required change I'd happily rebase again to master, as I always do. In the absence of any requested changes, I see no particular urgency. This might not be the best PR if you're curious about changes in code coverage, since I did not change any tests, or any infrastructure really. I just made things internal.

If you are curious about this, you can always of course check out my branch yourself and run the coverage numbers locally. *In principle* it shouldn't change much, since I did not change much infrastructure, just made it internal. (But still according to the report in its current form, I have a +3.29% change in coverage. Who can account for such things.)

**Comment by TomFinley** [comment]
> hurts my eyes :)

The goggles do nothing!

File: test/BaselineOutput/Common/EntryPoints/core_ep-list.tsv.bak [hunk 2]
```diff
@@ -0,0 +1,137 @@
+Data.CustomTextLoader	Import a dataset from a text file	Microsoft.ML.EntryPoints.ImportTextData	ImportText	Microsoft.ML.EntryPoints.ImportTextData+Input	Microsoft.ML.EntryPoints.ImportTextData+Output
```
Comment (Ivanidzo4ka) [inline test/BaselineOutput/Common/EntryPoints/core_ep-list.tsv.bak]: >CustomTextLoader [](start = 5, length = 16)

Why we need this file?

Comment (TomFinley) [reply test/BaselineOutput/Common/EntryPoints/core_ep-list.tsv.bak]: We don't, sorry. 😄 I'll remove it.

**Comment by eerhardt** [comment]
>     [BestFriend]

what does BestFriend on a public class mean?

---
Refers to: src/Microsoft.ML.HalLearners/OlsLinearRegression.cs:38 in 90096c7. [](commit_id = 90096c753cb0bb4c53b49b6877e0e068d69f1dcc, deletion_comment = False)

**Comment by eerhardt** [comment]
Did you mean to check this `core_ep-list.tsv.bak` file in?

File: src/Microsoft.ML.Data/EntryPoints/InputBase.cs [hunk 3]
```diff
@@ -37,10 +44,13 @@ public enum CachingOptions
     public abstract class LearnerInputBase
     {
         /// <summary>
-        /// The data to be used for training.
+        /// The data to be used for training. Used only in entry-points, since in the API the expected mechanism is
+        /// that the user iwll use the <see cref="IEstimator{TTransformer}.Fit(IDataView)"/> or some other train
```
Comment (eerhardt) [inline src/Microsoft.ML.Data/EntryPoints/InputBase.cs]: typeo- `iwll`

**Comment by TomFinley** [comment]
> > ```
> > [BestFriend]
> > ```
> 
> what does BestFriend on a public class mean?
> 
> Refers to: src/Microsoft.ML.HalLearners/OlsLinearRegression.cs:38 in [90096c7](https://github.com/dotnet/machinelearning/commit/90096c753cb0bb4c53b49b6877e0e068d69f1dcc). [](commit_id = [90096c7](https://github.com/dotnet/machinelearning/commit/90096c753cb0bb4c53b49b6877e0e068d69f1dcc), deletion_comment = False)

It means I'm silly. I'll fix that in next pass.

File: src/Microsoft.ML.Data/EntryPoints/InputBase.cs (hunk 3 above)
Comment (TomFinley) [reply src/Microsoft.ML.Data/EntryPoints/InputBase.cs]: > Did you mean to check this `core_ep-list.tsv.bak` file in?

No. @Ivanidzo4ka had the same comment, I think if you refresh the view you're using you should note that it is removed.

Comment (TomFinley) [reply src/Microsoft.ML.Data/EntryPoints/InputBase.cs]: > typeo- `iwll`

Whoops. Since internal only I will fix in next phase, if that's OK.

**Pull request merged**